tensorflow/python/ops/math_ops.py

```diff
@@ -1487,9 +1487,12 @@ def range(start, limit=None, delta=1, dtype=None, name="range"):  # pylint: disa
     start, limit = 0, start
 
   with ops.name_scope(name, "Range", [start, limit, delta]) as name:
-    start = ops.convert_to_tensor(start, dtype=dtype, name="start")
-    limit = ops.convert_to_tensor(limit, dtype=dtype, name="limit")
-    delta = ops.convert_to_tensor(delta, dtype=dtype, name="delta")
+    if not isinstance(start, ops.Tensor):
+      start = ops.convert_to_tensor(start, dtype=dtype, name="start")
+    if not isinstance(limit, ops.Tensor):
+      limit = ops.convert_to_tensor(limit, dtype=dtype, name="limit")
+    if not isinstance(delta, ops.Tensor):
+      delta = ops.convert_to_tensor(delta, dtype=dtype, name="delta")
```

Comment:
convert_to_tensor already is a no-op if the input is a tensor, so what is this doing?

Reply:
Thanks @alextp for the review. In #35710 the case was that the `dtype` passed along with `tf.range(start, limit, delta, dtype) ` is different from the tensor of the `start`/`limit`/`delta`. In other words, the following:
```
tf.range(tf.constant(4, dtype=tf.int32), dtype=tf.int64)
```

In that case, `convert_to_tensor` will try to convert `tf.constant(4, dtype=tf.int32)` to `dtype=tf.int64` and that is returning an error. That is the issue from  #35710 .